dataprep/clean/clean_url.py

```diff
@@ -0,0 +1,197 @@
+
+# removing auth params
+AUTH_VALUES = {"access_token", "auth_key", "auth", "password", "username", "login", "token"}
+
+# unified_list
+UNIFIED_AUTH_LIST = set()
+
+# STATs count
+STATS = {"cleaned": 0, "rows": 0}
+
+
+def clean_url(
+    df: Union[pd.DataFrame, dd.DataFrame],
+    column: str,
+    inplace: bool = False,
+    split: bool = False,
+    remove_auth: Union[bool, List[str]] = False,
+    report: bool = True,
+) -> Union[pd.DataFrame, dd.DataFrame]:
+
+    """
+    This function cleans url
+    Parameters
+    ----------
+    df
+        pandas or Dask DataFrame
+    column
+        column name
+    split
+        If True, split a column containing into the scheme, hostname, queries, cleaned_url columns
+        if set to False would return a new column of dictionaries with the relavant
+        information (scheme, host, etc.) in form of key-value pairs
+    inplace
+        If True, delete the given column with dirty data, else, create a new
+        column with cleaned data.
+    remove_auth
+        can be a bool, or list of string representing the names of Auth queries
+        to be removed. By default it is set to False
+    report:
+        Displays how many queries were removed from rows
+    """
+
+    # reset stats and convert to dask
+    reset_stats()
+    df = to_dask(df)
+
+    # unified list of auth removal params
+    if not isinstance(remove_auth, bool):
+        global UNIFIED_AUTH_LIST
+        UNIFIED_AUTH_LIST = {*AUTH_VALUES, *set(remove_auth)}
+
+    # specify the metadata for dask apply
+    meta = df.dtypes.to_dict()
+
+    if split:
+        meta.update(zip(("scheme", "host", "cleaned_url", "queries"), (str, str, str, str, str)))
+    else:
+        meta.update(zip(("url_details",), (str,)))
+
+    df = df.apply(
+        format_url,
+        args=(column, split, remove_auth),
+        axis=1,
+        meta=meta,
+    )
+
+    df = dask.compute(df, df.shape[0])[0]
+
+    if inplace:
+        if split:
+            df.drop(columns=[column])
+        else:
+            df.drop(columns=[column, "url_details"])
+
+    if report:
+        report_url(STATS)
+
+    return df
+
+
+def format_url(
+    row: pd.Series, column: str, split: bool, remove_auth: Union[bool, List[str]]
+) -> pd.Series:
+    """
+    This function formats each row of a pd.Series containing the url column
+    """
+
+    if split:
+        row["scheme"], row["host"], row["cleaned_url"], row["queries"] = get_url_params(
+            row[column], split=split, remove_auth=remove_auth
+        )
+
+    else:
+        val_dict = get_url_params(row[column], split=split, remove_auth=remove_auth)
+        row["url_details"] = val_dict
+    return row
+
+
+def get_url_params(url: str, split: bool, remove_auth: Union[bool, List[str]]) -> Any:
+    """
+    This function extracts all the params from a given url string
+    """
+    # regex for finding the query / params and values
+    re_queries = re.findall(QUERY_REGEX, url)
```

Comment:
When the input column contains non-string value, like `np.NAN`, it will not be parsed successfully and I encountered the error `TypeError: expected string or bytes-like object`. Can we have an error handling mechanism for that situation? Please refer to `clean_lat_long()` or `clean_email()` for the `error` parameter.

Reply:
Hi Yi, I have added a check there as well.